Update name to Declarative Automation Bundles (#148)
databricks/bundle-examples: knowledge_base/serverless_job/README.md

# Serverless job

> This example demonstrates how to define and use a serverless job in a Declarative Automation Bundle.

Reviewer: ```suggestion
This Declarative Automation Bundles example demonstrates how to define a job that runs on serverless compute.
```
